### PR TITLE
 Add data-plane-adoption job to the zuul dataplane operator github-check layout

### PR DESCRIPTION
**Pull request by marios**
This adds cifmw-data-plane-adoption-osp-17-to-extracted-crc so we get
adoption coverage in github-check.

https://issues.redhat.com/browse/OSPRH-5770

### CODE REVIEW

**Review by slagle — CHANGES_REQUESTED** [review]
I have concerns with adding this job, which takes 2.5 to 3 hours. Originally, our goal was 1 hour for check/gate jobs. We've grown to 1.5 hours organically with the crc jobs, and now up to 3 hours if this is added. This has a big effect on our velocity and estimation of how long it takes to get to dev complete.

I'd like to see further discussion on this.

File: zuul.d/projects.yaml
Comment (marios) [inline zuul.d/projects.yaml:9]: as discussed in yesterday's adoption call:

I'll add one of the jobs here explicitly instead of adding the data-plane-adoption-pipeline template. That way we will only run one job instead of both (the job with ceph, non tls will run here) and we can restrict when this job will run using files:

**Comment by marios** [comment]
> I have concerns with adding this job, which takes 2.5 to 3 hours. Originally, our goal was 1 hour for check/gate jobs. We've grown to 1.5 hours organically with the crc jobs, and now up to 3 hours if this is added. This has a big effect on our velocity and estimation of how long it takes to get to dev complete.
> 
> I'd like to see further discussion on this.



> I have concerns with adding this job, which takes 2.5 to 3 hours. Originally, our goal was 1 hour for check/gate jobs. We've grown to 1.5 hours organically with the crc jobs, and now up to 3 hours if this is added. This has a big effect on our velocity and estimation of how long it takes to get to dev complete.
> 
> I'd like to see further discussion on this.

as discussed yesterday, yes these jobs do take longer. It is currently around 2.5 hours (build history at [1] for convenience and the 3 hours applies to the new multinode job).

As mentioned yesterday I'll add just one job (not both) and we can target it with files so it will only run when necessary.

File: zuul.d/projects.yaml
Comment (slagle) [inline zuul.d/projects.yaml:15]: this is all of the code, so using files doesn't really help here. I don't think we want a 2.5 hour job on this repo.

Comment (marios) [reply zuul.d/projects.yaml:15]: well there is also /docs /examples /hack ;)

can we further target this though? Looking under those directories it was not clear to me that we could.

We need to check if changes proposed to this operator will break our understanding of the adoption workflow (to this point in time), as codified in the adoption tests run in this job.

Comment (slagle) [reply zuul.d/projects.yaml:15]: > We need to check if changes proposed to this operator will break our understanding of the adoption workflow (to this point in time), as codified in the adoption tests run in this job.

I don't think that running it on every PR matches some of the earlier agreed upon context setting for which jobs we should run where. Perhaps we can revisit those discussions, but I'd like it to be more broadly understood so there's some acknowledgement about the impact of this change.

Comment (marios) [reply zuul.d/projects.yaml:15]: the proposal is that it will not run on changes to /tests or /docs or the other directories.
as previously mentioned I didn't think we could make it more specific than I posted. However any coverage is better than no coverage so... can you propose an improvement to the files: list. Can/should we consider removing any of those? Can we be more specific with files and sub-directories? 

The idea is that we are constantly evolving our understanding of the adoption workflow, when new services are added (earlier on) or with new breaking features like the UpdateVersion resource and the latest work with OSPCIX-308. We have to update our "implementation" of the adoption workflow codified in the /tests/ subdirectory of the data-plane-adoption repo (and run in the adoption ci jobs).

It would be best to catch these breaking changes as close to the source as possible, which means gating on dataplane-operator and edpm-ansible.

We currently run adoption jobs in data-plane-adoption, ci-framework, install_yamls and openstack-operator repos. It seems to me that the omission of the actual dataplane related code is wrong.

Comment (slagle) [reply zuul.d/projects.yaml:15]: We've agreed to hold this PR for now, and then when dataplane-operator is merged into openstack-operator it will be covered by the adoption integration job. We can continue to refine with files/etc once that merge is complete.

Comment (marios) [reply zuul.d/projects.yaml:15]: ok. for right now and as also agreed, we will proceed with the edpm-ansible so at least that is more than nothing :D https://github.com/openstack-k8s-operators/edpm-ansible/pull/662

Closing this pull.

**Pull request closed without merging**
